aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,312 @@
+    glue.createDatabase(CreateDatabaseRequest.builder()
+        .catalogId(catalogId)
+        .databaseInput(IcebergToGlueConverter.toDatabaseInput(namespace, metadata))
+        .build());
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) throws NoSuchNamespaceException {
+    if (!namespace.isEmpty()) {
+      // if it is not a list all op, just check if the namespace exists and return itself.
+      if (namespaceExists(namespace)) {
+        return Lists.newArrayList(namespace);
+      }
+      throw new NoSuchNamespaceException(
+          "Glue does not support nested namespace, cannot list namespaces under %s", namespace);
+    }
+    // should be safe to list all before returning the list, instead of dynamically load the list.
+    String nextToken = null;
+    List<Namespace> results = new ArrayList<>();
+    while (true) {
+      GetDatabasesResponse response = glue.getDatabases(GetDatabasesRequest.builder()
+          .catalogId(catalogId)
+          .nextToken(nextToken)
+          .build());
+      if (response.hasDatabaseList()) {
+        results.addAll(response.databaseList().stream()
+            .map(GlueToIcebergConverter::toNamespace)
+            .collect(Collectors.toList()));
+        nextToken = response.nextToken();
+      }
+      if (nextToken == null) {
+        break;
+      }
+    }
+    return results;
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) throws NoSuchNamespaceException {
+    String databaseName = IcebergToGlueConverter.toDatabaseName(namespace);
+    try {
+      GetDatabaseResponse response = glue.getDatabase(GetDatabaseRequest.builder()
+          .catalogId(catalogId)
+          .name(databaseName)
+          .build());
+      return response.database().parameters();
+    } catch (InvalidInputException e) {
+      throw new NoSuchNamespaceException("invalid input for namespace %s, error message: %s",
+          namespace, e.getMessage());
+    } catch (EntityNotFoundException e) {
+      throw new NoSuchNamespaceException("fail to find Glue database for namespace %s, error message: %s",
+          databaseName, e.getMessage());
+    }
+  }
+
+  /**
+   * After completing this operation, you no longer have access to the tables
+   * (and all table versions and partitions that might belong to the tables)
+   * and the user-defined functions in the deleted database.
+   * AWS Glue deletes these "orphaned" resources asynchronously in a timely manner, at the discretion of the service.
+   * To ensure the immediate deletion of all related resources, before calling DeleteDatabase, use DeleteTableVersion
+   * or BatchDeleteTableVersion, DeletePartition or BatchDeletePartition, DeleteUserDefinedFunction,
+   * and DeleteTable or BatchDeleteTable, to delete any resources that belong to the database.
+   * @param namespace a namespace. {@link Namespace}
+   * @return always true
+   * @throws NamespaceNotEmptyException never thrown
```

Comment:
I'm a little nervous about this implementation not following the pattern of the other catalogs and refusing to drop if there is something in the namespace. Downstream code may be relying on this to prevent destructive operations ...

Reply:
Good point, this is definitely a different behavior between Glue and Hive. I can do a ListTable before dropping.